assets/js/components/dashboard-sharing/ModuleRecoveryAlert/index.js

```diff
@@ -128,10 +139,30 @@ export default function ModuleRecoveryAlert() {
 		await recoverModules( modulesToRecover );
 
 		setRecoveringModules( false );
-		setCheckboxes( null );
 
-		return { dismissOnCTAClick: false };
-	}, [ checkboxes, clearRecoveredModules, recoverModules ] );
+		// Dismiss if all recoverable modules are checked or only one module is available for recovery.
+		if (
+			( checkboxes !== null &&
+				recoverableModulesList.every(
+					( module ) => checkboxes[ module ]
+				) ) ||
+			recoverableModulesList.length === 1
+		) {
+			// Dismiss the notification with a short expiry to remove it from the queue immediately but allow
+			// it to be shown again if the issue reoccurs in future.
+			dismissNotification( id, {
+				expiresInSeconds: 10,
+				skipHidingFromQueue: false,
```

Comment:
`skipHidingFromQueue` is a bit confusing IMO, and since this is the default behavior I think it would be cleaner to omit here.

Reply:
Yes - we will get rid of this option once we separate out all notification components that serve multiple purposes - such as CTAs that show an AdminMenuTooltip. That issue will be fixed soon.